### DISCUSSION TITLE
external loop sample editing/time slicing script

### DISCUSSION THREAD

**Discussion by 777paranoia** [bangcorrupt/hacktribe#280] [Ideas]
I've used the E2 very extensively for the last six or so years as a drum machine sampler, but only very recently have I dig into the whole loop sampling/chopping features. I learned very quickly from using the OE2SSLE's unfinished time slicing utility that the E2's sample editing engine is quite powerful, and there is extremely limited resources when it comes to managing, editing, and even parsing the data being handled in regards to the Wave RIFF metadata chunks that the E2 interfaces with, especially if you're trying to address loop samples in bulk. Additionally, anyone using loop samples heavily knows the E2 is often prone to crashing or freezing when exporting time sliced samples, especially the longer they get, and it often results in corrupting your E2sSample.all file and losing all your work. 

Is there any way, at the bare minimum, to utilize the code or scripts that happen internally with setting time slices, "guessing" loop lengths, and setting step thresholds external from within the E2? So in theory, all of this could be done before loading the samples onto the E2? Idk if this is a long shot, but man I just spent 12+ hours trying to edit, slice, import, and export a batch of maybe 20 total loops with all the errors I encountered and would love to never spend that much time on it again lol. Thanks

**Comment by 777paranoia** [comment]
@bangcorrupt you see anything under the hood that could help with this?

**Comment by bangcorrupt** [comment]
> E2 is often prone to crashing or freezing when exporting time sliced samples

Can you share steps to reproduce this?  I haven't experienced it, but I don't use sample slicing very often.

I haven't written any tools for sample management yet, I think Oe2SSLE is the best option.  @untergeekDE  wrote some [notes on the RIFF chunk format](https://github.com/untergeekDE/electribe2-docs/blob/main/e2sSample-all-file-format.md), but I think the slice data comes after the WAVE file, with the sample patch data.

**Reply by 777paranoia** [reply]
Word. Does anyone reading this have the ability to modify Oe2SSLE and change some of the default values in the loop sample menu? Like perhaps setting the step/slices to all initialize at "off" and the slice sample count at "16" / "16" instead of "0" / "16"? Or to arrange the slice start/stop/attack/amp columns and the slice/step options at the bottom of the page to be next to each other left and right instead of above and below each other? Little things like this would make such a huge different, especially for Mac users dealing with all the sizing/target action bugs with the windows.

**Reply by 777paranoia** [reply]
*for Mac version 0.2.0

**Reply by bangcorrupt** [reply]
I probably won't work on the code for Oe2sSLE, but I started setting it up to run in a codespace.  See the [`codespace` branch of my fork](https://github.com/bangcorrupt/Oe2sSLE/tree/codespace).

- Click: Code > Codespaces > Create codespace on codespace
- It will take a while to load and start a docker container.  
- Click the 'Open in browser' button that appears in the bottom left.
- Once the desktop opens in a new tab, open a terminal window.
- Run `sudo apt update && sudo apt -y install python3-pyaudio python3-tk`
- Run `python3 Oe2sSLE_GUI.py`


Most of this could be baked into an image so the app loads straight away.

**Comment by BKLronin** [comment]
For info
I´ve lost one sampleset so far as it got stuck on saving. Not sure what time sliced means but I had sliced samples on there that were recorded on the machine itself. Didn´t happen since then mos track have a sliced sample form recordings.

**Comment by bangcorrupt** [comment]
> E2 is often prone to crashing or freezing when exporting time sliced samples

> I´ve lost one sampleset so far as it got stuck on saving.

Does this happen with the factory sampler firmware, or only with Hacktribe?
